wcfsetup/install/files/lib/data/user/rank/UserRankAction.class.php

```diff
@@ -24,4 +24,78 @@ class UserRankAction extends AbstractDatabaseObjectAction {
 	 * @inheritDoc
 	 */
 	protected $requireACP = ['delete'];
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function create() {
+		/** @var UserRank $rank */
+		$rank = parent::create();
+		
+		if (isset($this->parameters['rankImageFile']) && $this->parameters['rankImageFile']) {
+			if (!($this->parameters['rankImageFile'] instanceof UploadFile)) {
+				throw new \UnexpectedValueException("The parameter 'rankImageFile' must be an instance of ". UploadFile::class .".");
+			}
+			
+			if (!$this->parameters['rankImageFile']->isProcessed()) {
+				$fileName = $rank->rankID . '-' . $this->parameters['rankImageFile']->getFilename();
+				
+				rename($this->parameters['rankImageFile']->getLocation(), WCF_DIR . UserRank::RANK_IMAGE_DIR . $fileName);
+				$this->parameters['rankImageFile']->setProcessed(WCF_DIR . UserRank::RANK_IMAGE_DIR . $fileName);
+				
+				$updateData['rankImage'] = $fileName;
+				
+				$rankEditor = new UserRankEditor($rank);
+				$rankEditor->update($updateData);
+			}
+		}
+		
+		return $rank;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function update() {
+		parent::update();
```

Comment:
Should this be moved to the bottom so that you are able to update the database information by simply modifying `$this->parameters` instead of needing a separate call?